test/defaultbrowsercontext.spec.js

```diff
@@ -111,9 +111,9 @@ describe('launchPersistentContext()', function() {
       document.cookie = 'username=John Doe';
       return document.cookie;
     });
-    expect(documentCookie).toBe('username=John Doe');
     await page.waitForTimeout(2000);
     const allowsThirdParty = CHROMIUM || FFOX;
+    expect(documentCookie).toBe(allowsThirdParty ? 'username=John Doe' : '');
```

Comment:
Can you explain how moving this line which checks value of a constant variable helps with flakiness? Or did you mean to reevaluate it here?